### PR TITLE
Implementation of checking algorithm for companies

### PR DESCRIPTION
**Pull request by bjornarjatten**

### CODE REVIEW

**Review by emieboii — APPROVED** [review]

File: BDSA2020.Api/Controllers/CompanyRepositoryController.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                Util.LogError(e, isTest);
+                return StatusCode(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        [HttpDelete("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<bool>> Delete(Guid id, bool isTest = false)
```
Comment (simoboye) [inline BDSA2020.Api/Controllers/CompanyRepositoryController.cs:92]:
```suggestion
        [HttpDelete()]
        [Route("delete/{id}")]
        [ProducesResponseType(StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<bool>> Delete([FromRoute] Guid id, bool isTest = false)
```

File: BDSA2020.Api/Controllers/CompanyRepositoryController.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                Util.LogError(e, isTest);
+                return StatusCode(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        [HttpPut]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<bool>> Update([FromBody] UpdateCompanyDTO company, bool isTest = false)
```
Comment (simoboye) [inline BDSA2020.Api/Controllers/CompanyRepositoryController.cs:116]:
```suggestion
        [HttpPut]
        [Route("update")]
        [ProducesResponseType(StatusCodes.Status200OK)]
        [ProducesResponseType(StatusCodes.Status404NotFound)]
        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
        public async Task<ActionResult<bool>> Update([FromBody] UpdateCompanyDTO company, bool isTest = false)
```

File: BDSA2020.Entities/Context.cs [hunk 3]
```diff
@@ -40,12 +41,6 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
         protected override void OnModelCreating(ModelBuilder modelBuilder)
         {
             SetUpManyToManyKeys(modelBuilder);
-
-            modelBuilder
-                .Entity<Company>()
-                .HasMany(e => e.PlacementDescriptions)
-                .WithOne(e => e.Company);
-
```
Comment (simoboye) [inline BDSA2020.Entities/Context.cs:48]: As we talked about, we should some time make the relationship one to many once again. We can make this an issue in the issue tracker 💯

**Review by simoboye — APPROVED** [review]
👍

**Review by frederikw03 — APPROVED** [review]
:)

**Pull request merged**